RecoEgamma/EgammaTools/plugins/CalibratedElectronProducers.cc

```diff
@@ -0,0 +1,198 @@
+#ifndef RecoEgamma_EgammaTools_CalibratedElectronProducer_h
+#define RecoEgamma_EgammaTools_CalibratedElectronProducer_h
+
+//author: Alan Smithee
+//description: 
+//  this class allows the residual scale and smearing to be applied to electrons
+//  the scale and smearing is on the ecal part of the energy
+//  hence the E/p combination needs to be re-don, hence the E/p Combination Tools
+//  it re-applies the regression with the new corrected ecal energy
+//  returns a vector of calibrated energies and correction data, indexed by EGEnergySysIndex
+//  a port of EgammaAnalysis/ElectronTools/CalibratedElectronProducerRun2
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectronFwd.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+
+#include "CondFormats/DataRecord/interface/GBRDWrapperRcd.h"
+#include "CondFormats/EgammaObjects/interface/GBRForestD.h"
+#include "RecoEgamma/EgammaTools/interface/EpCombinationTool.h"
+#include "RecoEgamma/EgammaTools/interface/ElectronEnergyCalibrator.h"
+#include "RecoEgamma/EgammaTools/interface/EGEnergySysIndex.h"
+#include "RecoEgamma/EgammaTools/interface/EgammaRandomSeeds.h"
+
+#include "TRandom2.h"
+
+#include <vector>
+
+
+template<typename T>
+class CalibratedElectronProducerT: public edm::stream::EDProducer<>
```

Comment:
here and in the PhotonProducer please implement a meaningful fillDescriptions
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp

Reply:
sure. Will do it tomorrow

Reply:
done